### ISSUE TITLE
tests/manage/mcg/test_bucket_creation.py::TestBucketCreation::test_bucket_creation[3-CLI-DEFAULT-BACKINGSTORE]

### ISSUE THREAD

**Issue by achuzhoy** [red-hat-storage/ocs-ci#5815]
tests/manage/mcg/test_bucket_creation.py::TestBucketCreation::test_bucket_creation[3-CLI-DEFAULT-BACKINGSTORE]

Test bucket creation using the S3 SDK, OC command or MCG CLI. The factory checks the bucket's health by default.

OCP version 4.10.9
OC_OCS_VERSION: '4.9'

Steps to reproduce:
Run prep steps from https://ocs-ci.readthedocs.io/en/latest/usage.html

sed -i 's/ocs_csv_channel:.*/ocs_csv_channel: "stable-4.9"/' ocs_ci/framework/conf/default_config.yaml
sed -i 's/default_ocs_must_gather_latest_tag:.*/default_ocs_must_gather_latest_tag: "latest-4.9"/' ocs_ci/framework/conf/default_config.yaml
sed -i 's/ocs_version:.*/ocs_version: "4.9"/' ocs_ci/framework/conf/default_config.yaml

run-ci tests/ --html report.html -m acceptance --cluster-path /home/kni/ocs-ci/cluster/ --cluster-name spk-factory-0 --ocs-version 4.9 --ocp-version 4.10 --collect-logs --collect-logs-on-success-run --junit-xml junit.xml --ocsci-conf=ocs_ci/framework/conf/default_config.yaml --ocsci-conf=conf/deployment/baremetal/upi_1az_rhcos_nvme_intel_3m_3w.yaml

**Comment by petr-balogh** [comment]
```
sed -i 's/ocs_csv_channel:./ocs_csv_channel: "stable-4.9"/' ocs_ci/framework/conf/default_config.yaml
``` should not be needed if you use --ocs-version 4.9 then ocs-ci will load the ocs 4.9 config for you automatically:
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/framework/conf/ocs_version/ocs-4.9.yaml

Basically you should not edit default_config.yaml file at all.

If you need to overwrite some value you can create custom config file with that section/value and pass it via `--ocsci-conf ./path/to/custom/config.yaml`


I don't see rally much details about the failure of the test or what the issue is about?  You do not attach any logs or output from the test case if it's failing and hence no one can help here.

**Comment by achuzhoy** [comment]
@petr-balogh let me know if you need anything else.

**Comment by petr-balogh** [comment]
@red-hat-storage/red-squad  can you please take a look? As I pointed out in other issue, the better to put really whole log folder you set in config as it contains also debug level log files + must gather logs. Thanks

**Comment by MeridianExplorer** [comment]
This seems like a reproduction of a [known problem](https://bugzilla.redhat.com/show_bug.cgi?id=2062128).
When I compared the CLI's binary's hashes between existing 4.9.6-1 runs and the logs attached, I got different results, which lead me to suspect that an unsupported version was used in these runs.

Alexander - how was ODF installed on the tested clusters? Which precise version of ODF is used?

**Comment by achuzhoy** [comment]
@MeridianExplorer 


This is pretty much what I did:


git clone https://github.com/red-hat-storage/ocs-ci
cd ocs-ci
sudo yum install -y python38 libcurl-devel python38-devel openssl-devel libxml2-devel
python3.8 -m venv .venv

. .venv/bin/activate

pip install --upgrade pip setuptools

pip install -r requirements.txt
mkdir data
mkdir cluster/auth -p
cp ~/kubeconfig-spoke cluster/auth/kubeconfig

"place the kubeadmin password in" cluster/auth/kubeadmin-password

Update the pull-secret for pulling the must-gather image
sed -i 's/ocs_csv_channel:.*/ocs_csv_channel: "stable-4.9"/' ocs_ci/framework/conf/default_config.yaml
sed -i 's/default_ocs_must_gather_latest_tag:.*/default_ocs_must_gather_latest_tag: "latest-4.9"/' ocs_ci/framework/conf/default_config.yaml
sed -i 's/ocs_version:.*/ocs_version: "4.9"/' ocs_ci/framework/conf/default_config.yaml

run-ci tests/ --html report.html -m acceptance --cluster-path /home/kni/ocs-ci/cluster/ --cluster-name spk-factory-0 --ocs-version 4.9 --ocp-version 4.10 --collect-logs --collect-logs-on-success-run --junit-xml junit.xml --ocsci-conf=ocs_ci/framework/conf/default_config.yaml --ocsci-conf=conf/deployment/baremetal/upi_1az_rhcos_nvme_intel_3m_3w.yaml



What did I do wrong? :)

**Comment by petr-balogh** [comment]
as I pointed out in the comment above you should not edit default config, the --ocs-version and --odf-version will load automatically proper data to config.

Just out of the curiosity the ODF 4.9 and OCP 4.10 is what is currently installed on the cluster?

But as I see you used those params it should anyway overwrite those config values, so this will not be the case here I guess.

**Comment by MeridianExplorer** [comment]
Beyond the things Petr mentioned - it's not necessarily about what you've done locally, but rather about the way the clusters were installed and configured. Something with their images seems to be out of place.
The best thing I can ask for at the moment is access to one of the clusters where the problem happened. The next best thing being a must-gather collected by the tests.

**Comment by achuzhoy** [comment]
I tried earlier what Petr mentioned and saw the same results.
I'll reproduce and give you access (had to re-purpose the setup).

**Comment by achuzhoy** [comment]
Re-executed the test today:

The content of sasha.yaml:

REPORTING:
  default_ocs_must_gather_latest_tag: 'latest-4.9'
ENV_DATA:
  ocs_version: '4.9'
RUN:
  log_dir: "/tmp/sasha"
DEPLOYMENT:
  ocs_csv_channel: "stable-4.9"

#The executed command:
run-ci tests/ --html report.html -m acceptance --cluster-path /home/kni/ocs-ci/cluster/ --cluster-name spk-factory-0 --ocs-version 4.9 --ocp-version 4.10 --collect-logs --collect-logs-on-success-run --junit-xml junit.xml --ocsci-conf=ocs_ci/framework/conf/default_config.yaml --ocsci-conf=conf/deployment/baremetal/upi_1az_rhcos_nvme_intel_3m_3w.yaml  --ocsci-conf=/home/kni/ocs-ci/sasha.yaml|tee ocs_test.log




=========================== short test summary info ============================
FAILED tests/manage/mcg/test_bucket_creation.py::TestBucketCreation::test_bucket_creation[3-CLI-DEFAULT-BACKINGSTORE]
FAILED tests/manage/mcg/test_bucket_deletion.py::TestBucketDeletion::test_bucket_delete[3-CLI-DEFAULT-BACKINGSTORE]
FAILED tests/manage/pv_services/test_dynamic_pvc_accessmodes_with_reclaim_policies.py::TestDynamicPvc::test_rwo_dynamic_pvc[CephFileSystem-Delete]
ERROR tests/manage/pv_services/test_dynamic_pvc_accessmodes_with_reclaim_policies.py::TestDynamicPvc::test_rwo_dynamic_pvc[CephBlockPool-Retain]
FAILED tests/manage/pv_services/test_dynamic_pvc_accessmodes_with_reclaim_policies.py::TestDynamicPvc::test_rwo_dynamic_pvc[CephFileSystem-Delete]
ERROR tests/manage/pv_services/test_dynamic_pvc_accessmodes_with_reclaim_policies.py::TestDynamicPvc::test_rwx_dynamic_pvc[CephFileSystem-Retain]
= 3 failed, 9 passed, 21 skipped, 866 deselected, 3 errors in 4153.20s (1:09:13) =



All the output and logs are here: http://file.rdu.redhat.com/~achuzhoy/ocs/ocs_ci_run_result_may_5_2022.tgz

**Comment by achuzhoy** [comment]
Another test was carried out:
Cluster details:
(.venv) [kni@provisionhost-0-0 ocs-ci]$ oc get clusterversion
NAME      VERSION   AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.10.9    True        False         3d12h   Cluster version is 4.10.9
(.venv) [kni@provisionhost-0-0 ocs-ci]$ oc get co
NAME                                       VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
authentication                             4.10.9    True        False         False      18h     
baremetal                                  4.10.9    True        False         False      3d12h   
cloud-controller-manager                   4.10.9    True        False         False      3d12h   
cloud-credential                           4.10.9    True        False         False      3d12h   
cluster-autoscaler                         4.10.9    True        False         False      3d12h   
config-operator                            4.10.9    True        False         False      3d12h   
console                                    4.10.9    True        False         False      21h     
csi-snapshot-controller                    4.10.9    True        False         False      3d12h   
dns                                        4.10.9    True        False         False      3d12h   
etcd                                       4.10.9    True        False         False      3d12h   
image-registry                             4.10.9    True        False         False      3d12h   
ingress                                    4.10.9    True        False         False      21h     
insights                                   4.10.9    True        False         False      3d12h   
kube-apiserver                             4.10.9    True        False         False      3d12h   
kube-controller-manager                    4.10.9    True        False         False      3d12h   
kube-scheduler                             4.10.9    True        False         False      3d12h   
kube-storage-version-migrator              4.10.9    True        False         False      3d12h   
machine-api                                4.10.9    True        False         False      3d12h   
machine-approver                           4.10.9    True        False         False      3d12h   
machine-config                             4.10.9    True        False         False      3d12h   
marketplace                                4.10.9    True        False         False      3d12h   
monitoring                                 4.10.9    True        False         False      3d12h   
network                                    4.10.9    True        False         False      3d12h   
node-tuning                                4.10.9    True        False         False      3d11h   
openshift-apiserver                        4.10.9    True        False         False      21h     
openshift-controller-manager               4.10.9    True        False         False      3d12h   
openshift-samples                          4.10.9    True        False         False      3d12h   
operator-lifecycle-manager                 4.10.9    True        False         False      3d12h   
operator-lifecycle-manager-catalog         4.10.9    True        False         False      3d12h   
operator-lifecycle-manager-packageserver   4.10.9    True        False         False      21h     
service-ca                                 4.10.9    True        False         False      3d12h   
storage                                    4.10.9    True        False         False      3d12h 




The executed command:
 run-ci tests/ --html report.html -m acceptance --cluster-path /home/kni/ocs-ci/cluster/ --cluster-name spk-factory-0 --ocs-version 4.9 --ocp-version 4.10 --collect-logs --collect-logs-on-success-run --junit-xml junit.xml --ocsci-conf=ocs_ci/framework/conf/default_config.yaml --ocsci-conf=conf/deployment/baremetal/upi_1az_rhcos_nvme_intel_3m_3w.yaml --ocsci-conf=/home/kni/ocs-ci/sasha.yaml|tee ocs_test.log

content of sasha.yaml:
REPORTING:
  default_ocs_must_gather_latest_tag: 'latest-4.9'
ENV_DATA:
  ocs_version: '4.9'
RUN:
  log_dir: "/tmp/sasha"
DEPLOYMENT:
  ocs_csv_channel: "stable-4.9"




ceph status after the test:
sh-4.4$ ceph status
  cluster:
    id:     85f4e100-d57f-4c31-a5f8-e2d0052f19d3
    health: HEALTH_OK
 
  services:
    mon: 3 daemons, quorum a,b,c (age 18h)
    mgr: a(active, since 21h)
    mds: 1/1 daemons up, 1 hot standby
    osd: 9 osds: 9 up (since 21h), 9 in (since 3d)
    rgw: 1 daemon active (1 hosts, 1 zones)
 
  data:
    volumes: 1/1 healthy
    pools:   11 pools, 465 pgs
    objects: 63.06k objects, 139 GiB
    usage:   376 GiB used, 524 GiB / 900 GiB avail
    pgs:     465 active+clean
 
  io:
    client:   15 KiB/s rd, 15 KiB/s wr, 2 op/s rd, 3 op/s wr


Link to the log of the run:
http://file.rdu.redhat.com/~achuzhoy/ocs/ocs_ci_run_result_may_12_2022.tgz



TL;DR

=========================== short test summary info ============================
FAILED tests/manage/mcg/test_bucket_creation.py::TestBucketCreation::test_bucket_creation[3-CLI-DEFAULT-BACKINGSTORE]
FAILED tests/manage/mcg/test_bucket_deletion.py::TestBucketDeletion::test_bucket_delete[3-CLI-DEFAULT-BACKINGSTORE]
ERROR tests/manage/pv_services/test_pvc_assign_pod_node.py::TestPvcAssignPodNode::test_rwo_pvc_assign_pod_node[CephFileSystem]
= 2 failed, 21 passed, 10 skipped, 865 deselected, 1 error in 4425.89s (1:13:45) =
(.venv) [kni@provisionhost-0-0 ocs-ci]$

**Comment by MeridianExplorer** [comment]
I'm sorry, I did not notice the logs from 7 days ago.
I went over the May 5 and 12 logs, but sadly, neither contains the necessary information.
The OCS must-gather was not collected on May 5 (as can be seen in the run logs), and the May 12 logs do not contain any must-gather data at all.
This specific issue most likely happened due to a wrong `pull-secret`.

I believe that giving us access to a live cluster will be the easiest, most efficient method of debugging

**Comment by achuzhoy** [comment]
Reproduced yet again,

```=========================== short test summary info ============================
ERROR tests/manage/mcg/test_bucket_creation.py::TestBucketCreation::test_bucket_creation[3-S3-DEFAULT-BACKINGSTORE]                                                                                               
ERROR tests/manage/mcg/test_bucket_creation.py::TestBucketCreation::test_bucket_creation[3-OC-DEFAULT-BACKINGSTORE]                                                                                               
ERROR tests/manage/mcg/test_bucket_creation.py::TestBucketCreation::test_bucket_creation[3-CLI-DEFAULT-BACKINGSTORE]                                                                                              
ERROR tests/manage/mcg/test_bucket_deletion.py::TestBucketDeletion::test_bucket_delete[3-S3-DEFAULT-BACKINGSTORE]                                                                                                 
ERROR tests/manage/mcg/test_bucket_deletion.py::TestBucketDeletion::test_bucket_delete[3-CLI-DEFAULT-BACKINGSTORE]                                                                                                
ERROR tests/manage/mcg/test_bucket_deletion.py::TestBucketDeletion::test_bucket_delete[3-OC-DEFAULT-BACKINGSTORE]                                                                                                 
ERROR tests/manage/mcg/test_write_to_bucket.py::TestBucketIO::test_write_file_to_bucket[DEFAULT-BACKINGSTORE]                                                                                                     
===== 20 passed, 6 skipped, 873 deselected, 7 errors in 7201.23s (2:00:01) =====```

**Comment by achuzhoy** [comment]
to debug:
ssh ocp-edge36.lab.eng.tlv2.redhat.com  root/12345678  and run 'tmux a'

You'll be connected to a terminal with sourced kubeconfig,etc.
The output log is ocs_test.log

Note: We don't have too many setups, so we need the setup back as soon as possible.
Thanks!

**Comment by achuzhoy** [comment]
@MeridianExplorer  I ran a test with:
OCP: 4.10.20
ocs-operator.v4.10.4




=========================== short test summary info ============================
FAILED tests/manage/pv_services/test_pvc_assign_pod_node.py::TestPvcAssignPodNode::test_rwo_pvc_assign_pod_node[CephFileSystem]
FAILED tests/manage/pv_services/test_pvc_assign_pod_node.py::TestPvcAssignPodNode::test_rwx_pvc_assign_pod_node[CephFileSystem]
FAILED tests/manage/rgw/test_rgw_pod_existence.py::TestRGWPodExistence::test_rgw_pod_existence
ERROR tests/manage/mcg/test_bucket_creation.py::TestBucketCreation::test_bucket_creation[3-S3-DEFAULT-BACKINGSTORE]
ERROR tests/manage/mcg/test_bucket_creation.py::TestBucketCreation::test_bucket_creation[3-OC-DEFAULT-BACKINGSTORE]
ERROR tests/manage/mcg/test_bucket_creation.py::TestBucketCreation::test_bucket_creation[3-CLI-DEFAULT-BACKINGSTORE]
ERROR tests/manage/mcg/test_bucket_deletion.py::TestBucketDeletion::test_bucket_delete[3-S3-DEFAULT-BACKINGSTORE]
ERROR tests/manage/mcg/test_bucket_deletion.py::TestBucketDeletion::test_bucket_delete[3-CLI-DEFAULT-BACKINGSTORE]
ERROR tests/manage/mcg/test_bucket_deletion.py::TestBucketDeletion::test_bucket_delete[3-OC-DEFAULT-BACKINGSTORE]
ERROR tests/manage/mcg/test_write_to_bucket.py::TestBucketIO::test_write_file_to_bucket[DEFAULT-BACKINGSTORE]
ERROR tests/manage/pv_services/test_dynamic_pvc_accessmodes_with_reclaim_policies.py::TestDynamicPvc::test_rwo_dynamic_pvc[CephFileSystem-Delete]
ERROR tests/manage/pv_services/test_dynamic_pvc_accessmodes_with_reclaim_policies.py::TestDynamicPvc::test_rwx_dynamic_pvc[CephFileSystem-Delete]
FAILED tests/manage/pv_services/test_pvc_assign_pod_node.py::TestPvcAssignPodNode::test_rwo_pvc_assign_pod_node[CephFileSystem]
ERROR tests/manage/pv_services/pvc_clone/test_pvc_to_pvc_clone.py::TestClone::test_pvc_to_pvc_clone[CephFileSystem]
ERROR tests/manage/pv_services/pvc_resize/test_pvc_expansion.py::TestPvcExpand::test_pvc_expansion
ERROR tests/manage/pv_services/pvc_snapshot/test_pvc_snapshot.py::TestPvcSnapshot::test_pvc_snapshot[CephFileSystem]
= 3 failed, 13 passed, 7 skipped, 917 deselected, 13 errors in 6385.35s (1:46:25) =







The executed command was:
 run-ci tests/ --html report.html -m acceptance --cluster-path /home/kni/ocs-ci/cluster/ --cluster-name spk-factory-0 --ocs-version 4.10 --ocp-version 4.10 --collect-logs --collect-logs-on-success-run --junit-xml junit.xml --ocsci-conf=ocs_ci/framework/conf/default_config.yaml --ocsci-conf=conf/deployment/baremetal/upi_1az_rhcos_nvme_intel_3m_3w.yaml --ocsci-conf=/home/kni/ocs-ci/sasha.yaml|tee ocs_test.log


sasha.yaml content:
```
REPORTING:
  default_ocs_must_gather_latest_tag: 'latest-4.10'
ENV_DATA:
  ocs_version: '4.10'
RUN:
  log_dir: "/tmp/sasha"
DEPLOYMENT:
  ocs_csv_channel: "stable-4.10"
```

**Comment by achuzhoy** [comment]
@petr-balogh @MeridianExplorer 

Should we re-open this (closed) issue?
